torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -710,7 +710,22 @@ def triton_kernel_wrapper_mutation_dense(
                 element_size,
             )
 
-    kernel[grid_fn](**kwargs, **constant_args)
+    # move as many positional arguments from dicts to args as we
+    # can to circumvent the bug with the kwargs and pre_/post_hook:
+    # https://github.com/triton-lang/triton/issues/5082
+    # TODO: remove this when the Triton issue above is fixed
+    args = []
+    kwargs = kwargs.copy()
+    constant_args = constant_args.copy()
+    for name in kernel.arg_names:
+        if name in kwargs:
+            args.append(kwargs.pop(name))
+        elif name in constant_args:
+            args.append(constant_args.pop(name))
+        else:
+            break
```

Comment:
you can write this for loop better, you dont need copy/pop, just read indices?

Reply:
I need to pop, because I'm passing the `kwargs` and `constant_args` together with the collected `args`, no (that's also why I'm copying `kwargs` and `constant_args`, as I mutate the dicts). Otherwise, there will be duplicate argument passing in the line following this?